### ISSUE TITLE
cmake not working if built against libcx

### ISSUE THREAD

**Issue by SilvanScherrer** [bitwiseworks/libcx#28]
When I build cmake against libcx, the produced exe will not work as it should. The short debug I did was that select() gives back a "Invalid argument" message. There might be an issue with the select() overwrite. More debugging is needed to be 100% sure.

Just as a sidenote, cmake not built against libcx works of course as it should :)

**Comment by dmik** [comment]
This is fixed by 9f02a6af241e40cf545e7a33a3945f28f5319fd4.

**Comment by dmik** [comment]
Silvan claims there are still problems when building cmake with `-lcx` but there are different problems, not the originally reported problem with select. With the select issue fixed, Silvan gets this:
```
LIBC PANIC!!
_linitheap: sbrk failed!
pid=0x1bec ppid=0x1948 tid=0x0001 slot=0x00b0 pri=0x0200 mc=0x0000 ps=0x0017
E:\TREES\CMAKE\BUILD\BOOTSTRAP.CMK\CMAKE.EXE
Process dumping was disabled, use DUMPPROC / PROCDUMP to enable it.
CMake Error: Generator: execution of make failed. Make command was: "C:/USR/BIN/make.exe" "cmTC_28360/fast"
-- Checking whether CXX compiler has environ in stdlib.h - no
-- Checking whether CXX compiler has utimes
```
This happens when cmake.exe tries to start make.exe (which is actually kmk.exe in GNU make mode). The failure happens in kLIBC during initialisation of the default kLIBC heap (it tries to allocate 64k with DosAllocMem and the latter apparently fails for some reason). This needs a separate investigation and I don't think that it's directly related to LIBCx.

**Comment by dmik** [comment]
Just for the reference, this may be related to http://trac.netlabs.org/libc/ticket/345.

**Comment by dmik** [comment]
I'm reopening this as I found something new. It seems that the reason why it fails is LIBCx, not kLIBC. If I remove `_read` and `DosRead` overrides that work around the DosRead bug (see https://github.com/bitwiseworks/libcx/issues/21 for more info), all starts working smoothly, no more panic, cmake bootstrap/configure works and I guess build will work too. I need to investigate this.

**Comment by dmik** [comment]
The fix for the above problem is committed in 684bc119ca076dfd607d638ecc51fab0b0e71d1f. Closing this.
